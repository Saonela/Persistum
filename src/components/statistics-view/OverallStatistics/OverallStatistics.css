.overall-statistics__header {
    margin-bottom: var(--margin-l);
    font-size: var(--font-size-l);
    text-align: left;
}

.overall-statistics__list-header {
    display: grid;
    justify-content: center;
    grid-template-columns: 2fr 1fr 1fr 1fr;
    margin-bottom: var(--margin-m);
    font-size: var(--font-size-m);
}

.overall-statistics__list-header *:first-child {
    text-align: left;
}

.statistics-entry {
    display: grid;
    justify-content: center;
    align-items: center;
    grid-template-columns: 2fr 1fr 1fr 1fr;
    margin-bottom: var(--margin-s);
    transition: color 150ms ease-in;

}

.statistics-entry:hover {
    color: var(--color-theme);
}

.statistics-entry__activity {
    display: flex;
    align-items: center;
    text-align: left;
}

.statistics-entry .activity-color {
    margin-right: var(--margin-s);
}

@media only screen and (max-width: 820px) {
    .overall-statistics__list-header {
        display: none;
    }

    .overall-statistics__list {
        display: grid;
        grid-template-columns: 1fr 1fr;
        grid-column-gap: var(--margin-l);
    }

    .statistics-entry {
        grid-template-columns: 1fr;
        padding-bottom: var(--margin);
        margin-bottom: var(--margin);
        text-align: left;
    }

    .statistics-entry:not(:last-child) {
        border-bottom: 1px solid var(--color-border);
    }

    .statistics-entry * {
        padding-left: 0;
        margin-bottom: var(--margin-xs);
    }

    .statistics-entry > *:first-child{
        margin-bottom: var(--margin-s);
    }

    .statistics-entry > *:not(:first-child)::before {
        content: attr(data-label);
        display: inline-block;
        width: 110px;
    }
}

@media only screen and (max-width: 680px) {
    .overall-statistics {
        border: 0;
        border-radius: 0;
    }
}

@media only screen and (max-width: 540px) {
    .overall-statistics {
        padding-top: var(--padding-l);
        padding-right: var(--padding-m);
        padding-left: var(--padding-m);
    }

    .overall-statistics__list {
        grid-template-columns: 1fr;
    }
}

