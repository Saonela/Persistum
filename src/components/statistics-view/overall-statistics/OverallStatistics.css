.overall-statistics__list-header {
    display: grid;
    justify-content: center;
    grid-template-columns: 2fr 1fr 1fr 1fr;
    margin-bottom: var(--margin-m);
    font-size: var(--font-size-m);
}

.overall-statistics__list-header *:first-child {
    text-align: left;
}

.overall-statistics-entry {
    display: grid;
    justify-content: center;
    align-items: center;
    grid-template-columns: 2fr 1fr 1fr 1fr;
    margin-bottom: var(--margin-s);
    transition: color 150ms ease-in;
}

.overall-statistics-entry:hover {
    color: var(--color-theme);
}

@media only screen and (max-width: 820px) {
    .overall-statistics__list-header {
        display: none;
    }

    .overall-statistics__list {
        display: grid;
        grid-template-columns: 1fr 1fr;
        grid-column-gap: var(--margin-l);
    }

    .overall-statistics-entry {
        grid-template-columns: 1fr;
        padding-bottom: var(--margin);
        margin-bottom: var(--margin);
        text-align: left;
    }

    .overall-statistics-entry:not(:last-child) {
        border-bottom: 1px solid var(--color-border);
    }

    .overall-statistics-entry * {
        padding-left: 0;
        margin-bottom: var(--margin-xs);
    }

    .overall-statistics-entry > *:first-child {
        margin-bottom: var(--margin-s);
    }

    .overall-statistics-entry > *:not(:first-child)::before {
        content: attr(data-label);
        display: inline-block;
        width: 110px;
    }
}

@media only screen and (max-width: 680px) {
    .overall-statistics {
        border: 0;
        border-radius: 0;
    }
}

@media only screen and (max-width: 540px) {
    .overall-statistics {
        padding-top: var(--padding-l);
        padding-right: var(--padding-m);
        padding-left: var(--padding-m);
    }

    .overall-statistics__list {
        grid-template-columns: 1fr;
    }
}

