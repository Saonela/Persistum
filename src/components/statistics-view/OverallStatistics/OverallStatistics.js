import React from "react";
import "./OverallStatistics.css"
import ActivityColor from "../../shared/ActivityColor/ActivityColor";

function StatisticsView({activities, statistics}) {

    console.log('overall stats', statistics);

    return (
        <div className="app-panel app-border overall-statistics">
            <div className="overall-statistics__header">Overall</div>
            <div className="overall-statistics__list-header">
                <p className="overall-statistics__header-item">Activity</p>
                <p className="overall-statistics__header-item">Count</p>
                <p className="overall-statistics__header-item">First Occurrence</p>
                <p className="overall-statistics__header-item">Last Occurrence</p>
            </div>
            <div className="overall-statistics__list">
                {activities.map(activity =>
                    <div className="statistics-entry" key={activity.id}>
                        <p data-label="Activity" className="statistics-entry__activity">
                            <ActivityColor background={activity.style.background}/>
                            {activity.name}
                        </p>
                        <p data-label="Count" className="statistics-entry__count">{statistics[activity.id].count}</p>
                        <p data-label="First Occurrence" className="statistics-entry__first-timestamp">{statistics[activity.id].firstTimestamp}</p>
                        <p data-label="Last Occurrence" className="statistics-entry__last-timestamp">{statistics[activity.id].lastTimestamp}</p>
                    </div>
                )}
            </div>
        </div>
    )
}

export default StatisticsView;
