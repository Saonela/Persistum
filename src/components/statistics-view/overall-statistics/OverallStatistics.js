import React from "react";
import "./OverallStatistics.css"
import ActivityEntry from "../../shared/activity-entry/ActivityEntry";

function OverallStatistics({activities, statistics}) {
    return (
        <div className="app-panel app-border overall-statistics statistics-panel">
            <div className="statistics-panel__header">Overall</div>
            <div className="overall-statistics__list-header">
                <p className="overall-statistics__header-item">Activity</p>
                <p className="overall-statistics__header-item">Count</p>
                <p className="overall-statistics__header-item">First Occurrence</p>
                <p className="overall-statistics__header-item">Last Occurrence</p>
            </div>
            <div className="overall-statistics__list">
                {activities.map(activity =>
                    <div className="overall-statistics-entry" key={activity.id}>
                        <ActivityEntry activity={activity}/>
                        <p data-label="Count:" className="overall-statistics-entry__count">{statistics[activity.id].count}</p>
                        <p data-label="First Occurrence:" className="overall-statistics-entry__first-timestamp">{statistics[activity.id].firstTimestamp}</p>
                        <p data-label="Last Occurrence:" className="overall-statistics-entry__last-timestamp">{statistics[activity.id].lastTimestamp}</p>
                    </div>
                )}
            </div>
        </div>
    )
}

export default OverallStatistics;
