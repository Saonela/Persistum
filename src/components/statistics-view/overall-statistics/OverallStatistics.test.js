import OverallStatistics from "./OverallStatistics";
import {render, screen} from "@testing-library/react";
import React from "react";

const activities = [
    {
        id: 123456,
        name: 'Read 5 pages',
        completed: false,
        style: ''
    },
    {
        id: 999,
        name: 'Workout 30 minutes',
        completed: false,
        style: ''
    }
];

const statistics = {
    123456: {
        count: 2,
        firstTimestamp: '2010-11-02',
        lastTimestamp: '2020-11-05'
    },
    999: {
        count: 3,
        firstTimestamp: '2020-04-05',
        lastTimestamp: '2020-11-05'
    }
};

describe('OverallStatistics', () => {

    it('should display statistics', () => {
        render(<OverallStatistics activities={activities} statistics={statistics}/>);
        screen.getByText('Read 5 pages');
        screen.getByText('Workout 30 minutes');
    });

    it('should match snapshot', () => {
        const {asFragment} = render(<OverallStatistics activities={activities} statistics={statistics}/>);
        expect(asFragment()).toMatchSnapshot();
    });
});
