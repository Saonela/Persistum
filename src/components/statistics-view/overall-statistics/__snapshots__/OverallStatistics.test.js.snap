// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverallStatistics should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="app-panel app-border overall-statistics statistics-panel"
  >
    <div
      class="statistics-panel__header"
    >
      Overall
    </div>
    <div
      class="overall-statistics__list-header"
    >
      <p
        class="overall-statistics__header-item"
      >
        Activity
      </p>
      <p
        class="overall-statistics__header-item"
      >
        Count
      </p>
      <p
        class="overall-statistics__header-item"
      >
        First Occurrence
      </p>
      <p
        class="overall-statistics__header-item"
      >
        Last Occurrence
      </p>
    </div>
    <div
      class="overall-statistics__list"
    >
      <div
        class="overall-statistics-entry"
      >
        <div
          class="activity-entry"
          data-label="Activity"
        >
          <div
            aria-label="Activity color"
            class="activity-color"
            role="img"
          />
          Read 5 pages
        </div>
        <p
          class="overall-statistics-entry__count"
          data-label="Count:"
        >
          2
        </p>
        <p
          class="overall-statistics-entry__first-timestamp"
          data-label="First Occurrence:"
        >
          2010-11-02
        </p>
        <p
          class="overall-statistics-entry__last-timestamp"
          data-label="Last Occurrence:"
        >
          2020-11-05
        </p>
      </div>
      <div
        class="overall-statistics-entry"
      >
        <div
          class="activity-entry"
          data-label="Activity"
        >
          <div
            aria-label="Activity color"
            class="activity-color"
            role="img"
          />
          Workout 30 minutes
        </div>
        <p
          class="overall-statistics-entry__count"
          data-label="Count:"
        >
          3
        </p>
        <p
          class="overall-statistics-entry__first-timestamp"
          data-label="First Occurrence:"
        >
          2020-04-05
        </p>
        <p
          class="overall-statistics-entry__last-timestamp"
          data-label="Last Occurrence:"
        >
          2020-11-05
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
